CMakeLists.txt

```diff
@@ -161,6 +161,12 @@ option(COLORIZE_OUTPUT "Colorize output during compilation" ON)
 option(USE_ASAN "Use Address Sanitizer" OFF)
 option(USE_TSAN "Use Thread Sanitizer" OFF)
 option(USE_CUDA "Use CUDA" ON)
+# BUILD_SPLIT_CUDA must also be exported as an environment variable before building, with
+# `export BUILD_SPLIT_CUDA=1` because cpp_extension.py can only work properly if this variable
```

Comment:
it's unclear which keyword you want to use for user-setting ENV VAR name and which one to use for cmake flags. it's good to separate them instead of mixing them together with one name "BUILD_SPLIT_CUDA".

btw where is CUDA_SEPARABLE_COMPILATION set? is it an environment variable? and option?

Reply:
CUDA_SEPARABLE_COMPILATION is an option. Also, I thought using the same name is good because it's for the same purpose--I see that USE_CPP_COVERAGE is created in this way as well?

The purpose of my comment is to say that you can't only pass BUILD_SPLIT_CUDA into cmake with -D; setting the environment variable is necessary and sufficient. I thought creating a different name that represents the same thing would be more confusing.

Reply:
I see. so per Nikita's comment we dont need to explain this here b/c python setup script will correctly add them. maybe remove the comments about setting env var?

Reply:
I think both comment and conditions are fine.
Small nit: Please consider moving comment to the end of comment block rather than keeping it on the same line as ocmmand